### PR TITLE
Ops guides updates for monitoring and disk space

### PR DESCRIPTION
**Pull request by richardcloudsoft**
Some additions to the User Guide "Operations" section, covering starting, monitoring and stopping the server, and a discussion of its disk space usage.

### CODE REVIEW

File: guide/ops/requirements.md [hunk 1]
```diff
@@ -19,6 +19,43 @@ at least 8GB RAM and 100GB disk. The disk is just for logs, a small amount of pe
...
+
+* **Static files**: these are the Apache Brooklyn files themselves, plus
+  binaries for custom blueprints and integrations added to the `lib` directory.
+* **Persisted state**: when using [Persistence](persistence/index.html) -- which
+  is a prerequisite for [High Availability](high-availability.html) -- Brooklyn
+  will save data to a store location. Items in the persisted state include
+  metadata about the Brooklyn servers, catalog items, and metadata about all
+  running applications and entities.
+* **Log files**: Brooklyn writes detailed log files by default to its own
+  installation directory. This can be reconfigured to change the destination
```
Comment (drigodwin) [inline guide/ops/requirements.md]: I dont think this is true for the RPM or deb which install services - perhaps add something about where they put logs?

**Comment by rdowner** [comment]
How bizarre! Branch lives on the `rdowner` repository, but I inadvertently opened the PR under the `richardcloudsoft` user -- and GitHub let me do it??!! 😖

File: guide/ops/requirements.md [hunk 2]
```diff
@@ -19,6 +19,43 @@ at least 8GB RAM and 100GB disk. The disk is just for logs, a small amount of pe
...
+add to this - how much depends exactly on the entities involved and is therefore
+application specific, but as a guideline, a 3-node Riak cluster adds
+approximately 500KB to the persistence store.
+
+Log data can be a large consumer of disk space. By default Brooklyn generates
+two logfiles, one which logs notable information only, and another which logs at
+a debug level. Each logfile rotates when it hits a size of 100MB; a maximum of
+10 log files are retained for each type. The two logging streams combined,
+therefore, can consume up to 2GB of disk space. In the default configuration
+logs are saved to the Brooklyn installation directory. You will most likely want
```
Comment (drigodwin) [inline guide/ops/requirements.md]: This is a repeat of [above](https://github.com/apache/brooklyn-docs/pull/79#discussion_r67708812)

File: guide/ops/starting-stopping-monitoring.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+Server.  For information on using the Brooklyn Client CLI to access an already
+running Brooklyn Server, refer to [Client CLI Reference](cli/index.html).
+
+
+## Starting
+
+To launch Brooklyn, from the directory where Brooklyn is unpacked, run:
+
+{% highlight bash %}
+% bin/brooklyn launch > /dev/null 2>&1 & disown
```
Comment (drigodwin) [inline guide/ops/starting-stopping-monitoring.md]: Again this is not true when launched as a service

File: guide/ops/requirements.md [hunk 4]
```diff
@@ -19,6 +19,43 @@ at least 8GB RAM and 100GB disk. The disk is just for logs, a small amount of pe
 any binaries for custom blueprints/integrations.
 
 
+### Disk Space
+
+There are three main consumers of disk space:
+
+* **Static files**: these are the Apache Brooklyn files themselves, plus
```
Comment (aledsage) [inline guide/ops/requirements.md]: Perhaps reword: the installation and dependencies (e.g. Java). Therefore it would be applicable to RPM installs as well.

Separate point for blueprints added to the catalog. These can include: OSGi bundles for Java-based blueprints or associated resources; and/or jars added to the `lib/dropins/` directory for old-style Java-based blueprints.

File: guide/ops/requirements.md (hunk 1 above)
Comment (aledsage) [inline guide/ops/requirements.md]: Agreed. Perhaps we don't need to say here where it's written to - you've already linked to the logging section. Instead just say:

     **Log files**: Brooklyn writes info and debug log files. By default, these are 
    written to the local filesystem. This can be reconfigured to set the destination
    and to increase or decrease the detail in the logs. See the
    [Logging](logging.html) section for more details.

File: guide/ops/requirements.md [hunk 5]
```diff
@@ -19,6 +19,43 @@ at least 8GB RAM and 100GB disk. The disk is just for logs, a small amount of pe
...
+  metadata about the Brooklyn servers, catalog items, and metadata about all
+  running applications and entities.
+* **Log files**: Brooklyn writes detailed log files by default to its own
+  installation directory. This can be reconfigured to change the destination
+  files, and increase or decrease the detail of the logs. See the
+  [Logging](logging.html) page for more details.
+
+
+The Apache Brooklyn distribution itself, when unpacked, consumes approximately
+75MB of disk space. The space consumed by additional binaries for custom
```
Comment (aledsage) [inline guide/ops/requirements.md]: And associated dependencies (i.e. Java).

File: guide/ops/requirements.md [hunk 6]
```diff
@@ -19,6 +19,43 @@ at least 8GB RAM and 100GB disk. The disk is just for logs, a small amount of pe
...
+blueprints and integrations is application-specific.
+
+Persisted state, excluding catalog data, is relatively small, starting at
+approximately 300KB for a clean, idle Brooklyn server. Deploying blueprints will
+add to this - how much depends exactly on the entities involved and is therefore
+application specific, but as a guideline, a 3-node Riak cluster adds
+approximately 500KB to the persistence store.
+
+Log data can be a large consumer of disk space. By default Brooklyn generates
+two logfiles, one which logs notable information only, and another which logs at
```
Comment (aledsage) [inline guide/ops/requirements.md]: This has changed recently in master. We now write the jclouds wire log to a separate file.

File: guide/ops/requirements.md [hunk 7]
```diff
@@ -19,6 +19,43 @@ at least 8GB RAM and 100GB disk. The disk is just for logs, a small amount of pe
...
+a debug level. Each logfile rotates when it hits a size of 100MB; a maximum of
+10 log files are retained for each type. The two logging streams combined,
+therefore, can consume up to 2GB of disk space. In the default configuration
+logs are saved to the Brooklyn installation directory. You will most likely want
+to [reconfigure Brooklyn's logging](logging.html) to save logs to a location
```
Comment (aledsage) [inline guide/ops/requirements.md]: Perhaps we should mention some other production-quality logging technologies (e.g. syslog, logstash, splunk, etc).

While you're looking at this, can you double-check and fix this section: http://brooklyn.apache.org/v/latest/ops/logging.html#logging-aggregators. Does one really write to the syslog in order to feed the logs to logstash?! Although that's possible, I'd assume one uses a configuration more like that in https://www.elastic.co/guide/en/logstash/current/config-examples.html (but I'm not certain).

File: guide/ops/requirements.md [hunk 8]
```diff
@@ -19,6 +19,43 @@ at least 8GB RAM and 100GB disk. The disk is just for logs, a small amount of pe
...
+elsewhere, and to rotate logs according to your organisation's policy.
+
+
 ## Supported Operating Systems
 
 The recommended operating system is CentOS 6.x or RedHat 6.x.
```
Comment (aledsage) [inline guide/ops/requirements.md]: At some point we should add CentOS 7.x, but let's not do that yet.

Can you change the line below this to Ubuntu 14.04 please (rather than 12.04).

File: guide/ops/starting-stopping-monitoring.md (hunk 3 above)
Comment (aledsage) [inline guide/ops/starting-stopping-monitoring.md]: It feels like this bit is duplicating http://brooklyn.apache.org/v/latest/ops/server-cli-reference.html

Perhaps we should just reference that instead of having the starting/stopping sub-sections, and ensure that the server-cli-reference is good enough?

Comment (aledsage) [inline guide/ops/starting-stopping-monitoring.md]: Looking more closely, I see that the server-cli-reference doesn't have a good "stop" section. Where do you think is most appropriate to put your new section on stopping? (bearing in mind we'll need two variants: one for this approach, and one for rpm/deb packages).

File: guide/ops/starting-stopping-monitoring.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+This should lead to a fairly straightforward integration with many monitoring
+tools - the monitoring tool can discover the expected PID, and can execute the
+start or stop commands shown above as necessary.
+
+For example, here is a fragment of a `monitrc` file as used by [Monit](http://https://mmonit.com/monit/):
+
+{% highlight text %}
+check process apachebrooklyn with pidfile /opt/apache-brooklyn/pid_java
+    start program = "/bin/bash -c '/opt/apache-brooklyn/bin/brooklyn launch & disown'" with timeout 10 seconds
```
Comment (aledsage) [inline guide/ops/starting-stopping-monitoring.md]: Can we include the `--persist auto` here?

File: guide/ops/starting-stopping-monitoring.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    stop  program = "/bin/bash -c 'kill $( cat /opt/apache-brooklyn/pid_java )'"
```
Comment (aledsage) [inline guide/ops/starting-stopping-monitoring.md]: (Not necessary for this PR, unless there is a quick answer). What are the best practices for handling missing or stale pid files? Should we `rm` the file?

We'll need to add something about when it's running as a service, but that can be a separate PR.

**Comment by aledsage** [comment]
Looks good - a few minor comments.

We should see how this relates / overlaps with what @nakomis has been writing (which is mostly related to HA). In particular, your monitoring section could include the relevant REST api call for a basic health check.

File: guide/ops/requirements.md (hunk 6 above)
Comment (rdowner) [inline guide/ops/requirements.md]: @aledsage I've checked the master source code - I can see config files for the jclouds logging, but the only reference that includes the jclouds log config is commented out. Are you sure this is enabled by default?

File: guide/ops/starting-stopping-monitoring.md (hunk 3 above)
Comment (rdowner) [inline guide/ops/starting-stopping-monitoring.md]: @aledsage My opinion is that this page serves a different purpose to the CLI reference page and we should keep both. This is duplicating a just a few lines, and then sending the reader to the Server CLI page. I think this is the smallest possible amount of duplication that keeps both pages useful.

**Comment by rdowner** [comment]
@aledsage @drigodwin I've addressed your review comments.

File: guide/ops/requirements.md (hunk 6 above)
Comment (aledsage) [inline guide/ops/requirements.md]: You're right - I just talked to someone about doing this! It's not been done.

**Comment by aledsage** [comment]
LGTM; merging.
